pandas/core/arrays/sparse/array.py

```diff
@@ -805,13 +811,14 @@ def _get_val_at(self, loc):
         else:
             return libindex.get_value_at(self.sp_values, sp_loc)
 
-    def take(self, indices, allow_fill=False, fill_value=None):
+    def take(self, indices, allow_fill: bool = False, fill_value=None):
         if is_scalar(indices):
             raise ValueError(
                 "'indices' must be an array, not a scalar '{}'.".format(indices)
             )
         indices = np.asarray(indices, dtype=np.int32)
 
+        result: List
```

Comment:
not saying anything about contents?

Reply:
not until _take_with_fill and _take_without_fill are annotated. removed till then.